source/s3_request_messages.c

```diff
@@ -443,6 +477,53 @@ static int s_s3_message_util_add_content_range_header(
     return AWS_OP_SUCCESS;
 }
 
+/* Add a content-md5 header. */
+static int s_s3_message_util_add_content_md5_header(
+    struct aws_allocator *allocator,
+    struct aws_byte_buf *byte_buf,
+    struct aws_http_message *out_message) {
+
+    AWS_PRECONDITION(out_message);
+
+    // Compute MD5
+    struct aws_byte_cursor md5_input = aws_byte_cursor_from_buf(byte_buf);
+    uint8_t md5_output[AWS_MD5_LEN] = {0};
+    struct aws_byte_buf md5_output_buf = aws_byte_buf_from_array(md5_output, sizeof(md5_output));
```

Comment:
I think you can use aws_byte_buf_from_empty_array here and then you don't have to reset the len variable.